test/libsolidity/SemanticTest.cpp

```diff
@@ -362,9 +419,148 @@ void SemanticTest::printUpdatedSettings(ostream& _stream, string const& _linePre
 		_stream << _linePrefix << "// " << setting.first << ": " << setting.second << endl;
 }
 
+bool SemanticTest::checkLogs(TestFunctionCall& _call)
+{
+	_call.setLogs(SolidityExecutionFramework::recordedLogs());
+
+	TestFunctionCall* producer = _call.previousCall();
+	std::vector<FunctionCall const*> consumers{};
+	// Only non-builtins are able to produce logs.
+	// So lets search from the current call up to the first non-builtin.
+	while (producer != nullptr && producer->call().kind == FunctionCall::Kind::Builtin)
+		if (producer->previousCall() == nullptr)
+			break;
+		else
+		{
+			// On the way up to the producer we track all builtins that where on the way.
+			// Only builtins can consume logs, we store them in the consumers vector.
+			consumers.emplace_back(&producer->call());
+			producer = producer->previousCall();
+		}
+
+	// Producer will now point to the call that probably produced a log.
+	if (producer)
+	{
+		// We iterate through the consumers to find out what logs they have consumed.
+		for (auto& consumer: consumers)
+			for (auto logIdx: m_touchedLogs[consumer])
+				// All logs that where touched by the consumer, will be marked as
+				// touched within the producer.
+				touchLog(producer->call(), logIdx);
```

Comment:
I would say let's rather check that the number is the same and they are equal individually.